src/panels/lovelace/editor/config-elements/hui-tile-card-editor.ts

```diff
@@ -0,0 +1,150 @@
+import { HassEntity } from "home-assistant-js-websocket";
+import { html, LitElement, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import { assert, assign, object, optional, string } from "superstruct";
+import { COLORS } from "../../../../common/color/compute-color";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { computeDomain } from "../../../../common/entity/compute_domain";
+import { domainIcon } from "../../../../common/entity/domain_icon";
+import { capitalizeFirstLetter } from "../../../../common/string/capitalize-first-letter";
+import "../../../../components/ha-form/ha-form";
+import type { SchemaUnion } from "../../../../components/ha-form/types";
+import type { HomeAssistant } from "../../../../types";
+import type { TileCardConfig } from "../../cards/types";
+import type { LovelaceCardEditor } from "../../types";
+import { actionConfigStruct } from "../structs/action-struct";
+import { baseLovelaceCardConfig } from "../structs/base-card-struct";
+
+const cardConfigStruct = assign(
+  baseLovelaceCardConfig,
+  object({
+    entity: optional(string()),
+    name: optional(string()),
+    icon: optional(string()),
+    color: optional(string()),
+    tap_action: optional(actionConfigStruct),
+    icon_tap_action: optional(actionConfigStruct),
+  })
+);
+
+@customElement("hui-tile-card-editor")
+export class HuiTileCardEditor
+  extends LitElement
+  implements LovelaceCardEditor
+{
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: TileCardConfig;
+
+  public setConfig(config: TileCardConfig): void {
+    assert(config, cardConfigStruct);
+    this._config = config;
+  }
+
+  private _schema = memoizeOne(
+    (entity: string, icon?: string, entityState?: HassEntity) =>
+      [
+        { name: "entity", selector: { entity: {} } },
```

Comment:
We should really only promote entity. Using the default ha-form makes all the same priority. Name and icon we really prefer at entity level so it's ok everywhere, and tap action defaults are solid too.

Reply:
I agree, we need to do some work for the editor to have primary settings and advanced settings.
I added lot of options in Mushroom. People like it but there is not hierarchy/priority between fields.

Reply:
You don't have to address it in this PR, but let's not forget about it.